packages/components/src/InputValidation/InputValidation.tsx

```diff
@@ -0,0 +1,75 @@
+import React from "react";
+import { AnimatePresence, motion } from "framer-motion";
+import { Text } from "../Text";
+import styles from "./InputValidation.css";
+
+type ValidationStatus = "success" | "error" | "warn" | "info";
+
+export interface ValidationProps {
+  /**
+   * Defines the status of the validation message. This also determines how the
+   * message is styled in the UI.
+   */
+  status: ValidationStatus;
+
+  /**
+   * The message that gets shown to the user.
+   */
+  message: string;
+
+  /**
+   * Determines if the `message` shows up for the user.
+   */
+  shouldShow?: boolean;
+}
+
+interface InputValidationProps {
+  /**
+   * Array of validation messages
+   */
+  messages: ValidationProps[];
+}
+
+interface StatusMap {
+  [key: string]: ValidationStatus;
```

Comment:
Perhaps something like…

```suggestion
  [key: keyof ValidationStatus]: ValidationStatus;
```

I don't think that is quite the right syntax.

Reply:
@seanhealy funny thing about this... I actually don't need this anymore because it maps directly to text variations now

Reply:
there also seems to be problems with what you suggested tho https://github.com/microsoft/TypeScript/issues/5683